src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -15,36 +16,50 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
         Type = parameter.Type;
         Name = parameter.Name;
         Source = EndpointParameterSource.Unknown;
+        HandlerArgument = $"{parameter.Name}_local";
 
-        if (GetSpecialTypeCallingCode(Type, wellKnownTypes) is string callingCode)
+        var fromQueryMetadataInterfaceType = wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromQueryMetadata);
+
+        if (GetSpecialTypeAssigningCode(Type, wellKnownTypes) is string assigningCode)
         {
             Source = EndpointParameterSource.SpecialType;
-            CallingCode = callingCode;
+            AssigningCode = assigningCode;
+        }
+        else if (parameter.HasAttributeImplementingInterface(fromQueryMetadataInterfaceType))
+        {
+            Source = EndpointParameterSource.Query;
+            AssigningCode = $"httpContext.Request.Query[\"{parameter.Name}\"]";
         }
+        else
+        {
+            // TODO: Inferencing rules go here - but for now:
+            Source = EndpointParameterSource.Unknown;
+        }
+
+        if (parameter.Type is INamedTypeSymbol parameterType && parameterType.ContainingType?.SpecialType == SpecialType.System_Nullable_T)
```

Comment:
```suggestion
        if (parameter.Type is INamedTypeSymbol parameterType && parameterType.NullableAnnotation == NullableAnnotation.Annotated)
```

Also, I would stick the logic for setting `IsOptional` inside the case for the parameter itself. There's enough subtleties between optionality for different parameter types:

- Body parameters needs to handle the `AllowEmpty` property
- `BindAsync` parameters need to handle distinctions in nullability annotation between the parameter and the value returned from `BindAsync`